protocol/clproto_cpp/test/tests/test_trajectory_proto.cpp

```diff
@@ -0,0 +1,54 @@
+#include <chrono>
+#include <gtest/gtest.h>
+#include <string>
+
+#include <state_representation/trajectory/CartesianTrajectory.hpp>
+#include <state_representation/trajectory/JointTrajectory.hpp>
+
+#include "clproto.hpp"
+#include "include/test_encode_decode.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+
+using namespace state_representation;
+
+template<typename T>
+static void test_trajectory_equal(const T& send_state, const T& recv_state) {
+  EXPECT_EQ(send_state.get_type(), recv_state.get_type());
+  EXPECT_STREQ(send_state.get_name().c_str(), recv_state.get_name().c_str());
+  if constexpr (std::is_same_v<T, CartesianTrajectory>) {
+    EXPECT_STREQ(send_state.get_reference_frame().c_str(), recv_state.get_reference_frame().c_str());
+  } else {
+    EXPECT_EQ(send_state.get_joint_names(), recv_state.get_joint_names());
+  }
+  EXPECT_EQ(send_state.get_size(), recv_state.get_size());
```

Comment:
to avoid out_of_range exception:
```suggestion
  ASSERT_EQ(send_state.get_size(), recv_state.get_size());
```